engine/apps/api/views/alert_group.py

```diff
@@ -165,9 +166,13 @@ def filter_by_involved_users(self, queryset, name, value):
         if not users:
             return queryset
 
+        alert_groups_by_users_qs = (
+            UserNotificationPolicyLogRecord.objects.filter(author__in=users).distinct().only("id").all()
+        )
+
         queryset = queryset.filter(
             # user was notified
-            Q(personal_log_records__author__in=users)
+            Q(id__in=alert_groups_by_users_qs)
```

Comment:
Trying using subquery instead of join https://docs.djangoproject.com/en/4.2/ref/models/querysets/#in